core/src/main/scala/api/attributes.scala

```diff
@@ -0,0 +1,40 @@
+package sfs
+package api
+
+object attributes {
+
+  final case class Permissions(
+    ownerRead: Boolean, ownerWrite: Boolean, ownerExecute: Boolean,
+    groupRead: Boolean, groupWrite: Boolean, groupExecute: Boolean,
+    otherRead: Boolean, otherWrite: Boolean, otherExecute: Boolean
+  )
+  object Permissions {
+    implicit def empty: api.Empty[Permissions] =
+      api.Empty(Permissions(false, false, false, false, false, false, false, false, false))
+  }
```

Comment:
I would love to put it in. The problem here is that I would cut them up and use them on two sides of the abstraction. So the question is, what to put in the middle, in `api.attributes`?

Hmm, thinking about it, it would be `UnixPerms`. I'll take a look in a minute to see how it fits in.